src/GovUk.Education.ExploreEducationStatistics.Data.Services.Tests/ReleaseServiceTests.cs

```diff
@@ -401,13 +449,139 @@ await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbC
 
-                Assert.Equal(contentRelease.Id, result.Right.ReleaseId);
+                Assert.Equal(contentRelease.Id, result.Right.Id);
 
                 var subjects = result.Right.Subjects;
 
                 Assert.Single(subjects);
                 Assert.Equal(releaseSubject2.Subject.Id, subjects[0].Id);
-                Assert.Equal(releaseSubject2.Subject.Name, subjects[0].Label);
+                Assert.Equal(releaseSubject2.Subject.Name, subjects[0].Name);
+            }
+        }
+
+        [Fact]
+        public async Task GetRelease_FiltersSubjectsWithNoImport()
+        {
+            var releaseId = Guid.NewGuid();
+
+            var contentRelease = new Release
+            {
+                Id = releaseId
+            };
+
+            var statisticsRelease = new Data.Model.Release
+            {
+                Id = releaseId
+            };
+
+            var releaseSubject = new ReleaseSubject
+            {
+                Release = statisticsRelease,
+                Subject = new Subject
+                {
+                    Id = Guid.NewGuid(),
+                    Name = "Subject 1"
+                }
+            };
+
+            var releaseFile = new ReleaseFile
+            {
+                Release = contentRelease,
+                File = new File
+                {
+                    Release = contentRelease,
+                    Filename = "data1.csv",
+                    Type = FileType.Data,
+                    SubjectId = releaseSubject.Subject.Id
+                }
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            var statisticsDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddAsync(contentRelease);
+                await contentDbContext.AddRangeAsync(releaseFile);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                await statisticsDbContext.AddRangeAsync(releaseSubject);
+                await statisticsDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                var service = BuildReleaseService(
+                    contentDbContext: contentDbContext,
+                    statisticsDbContext: statisticsDbContext
+                );
+
+                var result = await service.GetRelease(contentRelease.Id);
+
+                Assert.True(result.IsRight);
+
+                Assert.Equal(contentRelease.Id, result.Right.Id);
+
+                var subjects = result.Right.Subjects;
+
+                Assert.Empty(subjects);
```

Comment:
I think we might need to fix up some of the tickets in epic EES-1898 _Data inconsistencies_ before this goes onto Production like 1900, 1901, and because this method also filters out non-complete imports, possibly 1897 too, depending on whether these are all the latest published Releases or not. They might not be anymore.